drivers/can/can_xlnx.h

```diff
@@ -0,0 +1,385 @@
+/*
+ * Xilinx Processor System CAN controller driver
+ *
+ * Driver private data declarations
+ *
+ * Copyright (c) 2022, Weidmueller Interface GmbH & Co. KG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _ZEPHYR_DRIVERS_CAN_CAN_XLNX_H_
+#define _ZEPHYR_DRIVERS_CAN_CAN_XLNX_H_
+
+#include <stdint.h>
+
+/* The Xilinx CAN controller always has 4 acceptance filters */
+#define CAN_XLNX_NUM_ACCEPTANCE_FILTERS	4
+/* The RX/TX FIFOs of the Xilinx CAN controller are always 64 entries deep */
+#define CAN_XLNX_RX_TX_FIFO_DEPTH	64
```

Comment:
I'm disappointed to see (from the Z7K TRM) that the Xilinx PS7 CAN IP TX FIFO suffers from priority inversion. Even though having a nice HW FIFO available, it's no good when its sorting does not adhere to the priority of the CAN frames contained in it (it would need a HW priority queue instead).

I'm afraid using the HW FIFO here is not acceptable as it can lead to a high-priority frame having to wait for up to 63 lower-priority frames to be sent before itself.

I am slowly making progress on a software priority queue for use in CAN drivers where the underlying hardware has this "feature" (or misses this feature, has this bug, ...), but it is not ready yet. For now I suggest limiting this driver to use only _one_ position in the HW TX FIFO at a time. Sorry.